app/src/org/commcare/adapters/EntityListAdapter.java

```diff
@@ -276,8 +290,44 @@ public boolean isEmpty() {
         return getCount() > 0;
     }
 
-    public void applyFilter(String s) {
-        filterValues(s);
+    public synchronized void filterByString(String filterRaw) {
+        if (entitySearcher != null) {
+            entitySearcher.cancelSearch();
+        }
+        // split by whitespace
+        String[] searchTerms = filterRaw.split("\\s+");
+        for (int i = 0; i < searchTerms.length; ++i) {
+            searchTerms[i] = StringUtils.normalize(searchTerms[i]);
+        }
+        currentSearchTerms = searchTerms;
+        searchQuery = filterRaw;
+        entitySearcher =
+                new EntityStringFilterer(this, searchTerms, mAsyncMode,
+                        mFuzzySearchEnabled, mNodeFactory, full, context);
+        entitySearcher.start();
+    }
+
+    /**
+     * Filter entity list to only include entities that have extra keys present
+     * in the provided mapping.  Reorders entities by the key ordering of the
+     * mapping.
+     */
+    public synchronized void filterByKeyedCalloutData(OrderedHashtable<String, String> keyToExtraDataMapping) {
+        externalData = keyToExtraDataMapping;
+
+        if (entitySearcher != null) {
+            entitySearcher.cancelSearch();
+        }
+        LinkedHashSet<String> topMatchingCaseIds = new LinkedHashSet<>();
```

Comment:
This variable name seems more specific to the fingerprint scanning use case than the rest of the code. I don't necessarily object because it helped me understand better what was going on, but it seems like you were aiming to keep things generic to other use cases, so it's a little odd for this one variable to be inconsistent with that.